dvc/cachemgr.py

```diff
@@ -28,16 +30,29 @@ def __init__(self, repo):
         self.config = config = repo.config["cache"]
         self._odb = {}
 
+        default = None
+        if repo and repo.dvc_dir and not isinstance(repo.fs, GitFileSystem):
+            default = repo.fs.path.join(repo.dvc_dir, self.CACHE_DIR)
+        elif repo:
+            dvc_dir = os.path.join(
+                repo.scm.root_dir,
+                *repo.fs.path.relparts(repo.root_dir, "/"),
+                repo.DVC_DIR,
+            )
+            if os.path.exists(dvc_dir):
+                default = os.path.join(dvc_dir, self.CACHE_DIR)
+
```

Comment:
Not the best look and there are lots of `isinstance(repo.fs, GitFileSystem)` across the codebase elsewhere, but the proper solution will have to wait. Will probably get back to this when messing with `repo.tmp_dir` for persistent index.

Reply:
I find the whole `Repo()` constructor taking a "git" filesystem to be a wrong approach, needlessly complicating things. It should require a physical filesystem during instantiation.

Reply:
@skshetry That would not cover bare repos. We need to be able to operate on virtual filesystems.

Reply:
We just need to clearly handle "workspace-fs" (e.g. where the scm is, odb is, etc) from "vfs" (where dvcfiles are, some configs are, etc). It will get better from here.

Reply:
Bare repos aren’t that important, we could easily drop support (or, clone if needed).

I am not saying dvc not to support virtual filesystems, but loading config always requires a working filesystem. So, `Repo.__init__` could always require a physical filesystem.

The other cases could be handled through different classmethods: `Repo.from_git()` etc.

Reply:
Ah, re-read your message. Yeah, I agree, the alternative is to just pass a rev and scm or something. Also looked at it a bit. Might get back to it later (external_repo is also annoying).